kt/godot-coroutine-library/src/main/kotlin/godot/coroutines/awaitResourceLoad.kt

```diff
@@ -37,51 +39,13 @@ public suspend inline fun ResourceLoader.awaitLoad(
         return this.load(path)
     }
 
-    val error = this.loadThreadedRequest(
-        path = path,
-        typeHint = typeHint,
-        useSubThreads = useSubThreads,
-        cacheMode = cacheMode,
-    )
-
-    if (error != Error.OK) {
-        GD.printErr("Could not trigger resource load. Got error: $error")
-        return null
+    // Start a new job so we have a suspension point in case the coroutine is currently in the main thread.
+    val job = GodotCoroutine.launch(Dispatchers.Default) {
+        load(path)
     }
+    job.join()
```

Comment:
this could be made `async` instead of launch and thus directly return the result instead of calling load 2 times